components/editable-description.tsx

```diff
@@ -0,0 +1,93 @@
+"use client";
+
+import { useState } from "react";
+import { useRouter } from "next/navigation";
+
+import { updateProject } from "@/lib/actions";
+import { DesignProject } from "@/lib/types";
+
+import { CardContent, CardHeader, CardDescription } from "@/components/ui/card";
+import { Textarea } from "@/components/ui/textarea";
+import { toast } from "@/components/ui/use-toast";
+import { Button } from "./ui/button";
+
+export default function EditableDescription({
+  project,
+}: {
+  project: DesignProject;
+}) {
+  const router = useRouter();
+
+  const [description, setDescription] = useState(
+    project.description || "Untitled",
+  );
+  const [originalDescription, setOriginalDescription] = useState(
+    project.description || "Untitled",
+  );
+  const [editing, setEditing] = useState(false);
+
+  async function clickUpdateProject() {
+    try {
+      const updates = { description: description };
+      const result = await updateProject(project.id, updates);
+      setOriginalDescription(result?.description || description);
+      setEditing(false);
+      toast({ description: "Project updated succesfully." });
+      router.refresh();
+    } catch (error) {
+      toast({ description: "Project failed to be updated." });
+    }
+  }
+
+  function cancelEditing() {
+    setDescription(originalDescription); // Revert to original title
+    setEditing(false);
+  }
+
+  return (
+    <>
+      <CardHeader>
+        <div className="flex items-center gap-1">
```

Comment:
justify-between and the edit button variant secondary
I think it looks a bit better that way :)